---
title: "ofav_wgcna_revisions"
author: "Allyson DeMerlis"
date: "5/6/2021"
output: html_document
---

###Install and Load Packages
```{r}
library(tidyverse)
library(DESeq2)
library(ggrepel)
library(apeglm)
library(EnhancedVolcano)
library(ComplexHeatmap)
library(pheatmap)
library(genefilter)
library(WGCNA)
library(flashClust)
library(edgeR) #counts per million
library(reshape) #melt
library(clusterProfiler)
library(here)
```

```{r , include = F}
#ofav
load("../rDATA_objects_for_analysis/ofav_annotations_complete_4analysis.RData")
load("../rDATA_objects_for_analysis/ofav_counts_filt_4analysis.RData")
load("../rDATA_objects_for_analysis/ofav_metadata_4analysis.RData")
```

### DDS Object for WGCNA
```{r}
dds_ofav <- DESeqDataSetFromMatrix(countData=cccall_ofav, colData=ofav_metadata, design= ~ Experiment + Correct_treatment)

nrow(dds_ofav) #18066 genes
ncol(dds_ofav) #26 samples

keep_90 <- rowSums(counts(dds_ofav) >= 10) > (nrow(ofav_metadata)*0.9)
dds_ofav_wgcna <- dds_ofav[keep_90, ]
table(keep_90)
# 4,301 genes with counts over 10 in 90% of samples - useful for WGCNA because we need to subset them... can't use all 13,000+ genes
```


```{r, Genes for WGCNA}
# The following setting is important, do not omit.
options(stringsAsFactors = FALSE)

# Normalize expression data for WGCNA using VST tranformation
vsd_ofav_wgcna <- vst(dds_ofav_wgcna, blind = FALSE) #blind = TRUE does not take into account the conditions set in dds model. you want the corrections for WGCNA, so set to FALSE
assay(vsd_ofav_wgcna) <- limma::removeBatchEffect(assay(vsd_ofav_wgcna), vsd_ofav_wgcna$Experiment)
#remove batch effect

# obtain vst-transformed counts
countdata_vst_ofav <- assay(vsd_ofav_wgcna)

# Convert to matrix and transpose, check dimensions
datExpr_ofav <- t(countdata_vst_ofav) #because WGCNA expects a certain column/row input type
dim(datExpr_ofav)
#26 samples, 4,301 genes...
```


```{r, plot to look for outliers initially}
gsg = goodSamplesGenes(datExpr_ofav, verbose = 3);
gsg$allOK #TRUE = all genes have passed the cuts yay

sampleTree = hclust(dist(datExpr_ofav), method = "average");
# Plot the sample tree: Open a graphic output window of size 12 by 9 inches
# The user should change the dimensions if the window is too large or too small.
sizeGrWindow(12,9)
#pdf(file = "Plots/sampleClustering.pdf", width = 12, height = 9);
par(cex = 0.6);
par(mar = c(0,4,2,0))
plot(sampleTree, main = "Sample clustering to detect outliers", sub="", xlab="", cex.lab = 1.5,
cex.axis = 1.5, cex.main = 2)
```


```{r, Soft Threshold}
#Soft Threshold check - choosing which nodes of tree to prune out. Strength connection of what you want to keep. Trim out low correlations, because it is trying to make connections with each gene.

#running a heuristic to see what is the best choice for most genes. that's why it's a soft threshold, not a hard one. 

# Find correlation power R^N that satisfies scale free critereon (SFT.R.sq>0.9)
sft_ofav <- pickSoftThreshold(datExpr_ofav, verbose=5)

# Plot the results:
powers = c(c(1:10), seq(from = 12, to=20, by=2))
sizeGrWindow(9, 5)
par(mfrow = c(1,2));
cex1 = 0.9;
# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft_ofav$fitIndices[,1], -sign(sft_ofav$fitIndices[,3])*sft_ofav$fitIndices[,2],
xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
main = paste("Scale independence"));
text(sft_ofav$fitIndices[,1], -sign(sft_ofav$fitIndices[,3])*sft_ofav$fitIndices[,2],
labels=powers,cex=cex1,col="red");
# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red")
# Mean connectivity as a function of the soft-thresholding power
plot(sft_ofav$fitIndices[,1], sft_ofav$fitIndices[,5],
xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
main = paste("Mean connectivity"))
text(sft_ofav$fitIndices[,1], sft_ofav$fitIndices[,5], labels=powers, cex=cex1,col="red")
```

```{r, Signed Adjacency Matrix}
# This follows the tutorial: 
# https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/FemaleLiver-02-networkConstr-man.pdf
# 
#From Mike Connelly:
# I have chosen the following network construction parameters for the following reasons:
# First, following the recommendations of the WGCNA developers (https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/faq.html), a signed network was chosen to be able to detect positive and negative gene correlations, and the biweight midcorrelation was used since it is more robust to outliers. 

#adjacency matrix = how close together the nodes are
adjacency <- adjacency(datExpr_ofav,
      # Network construction arguments:  correlation, adjacency function,  and topological overlap map options
                       corFnc = "bicor", # midweight bi-correlation (more robust to outliers than pearson correlation)
                       power = 20, #WHAT SHOULD THIS BE
                       type = "signed") # signed - it knows down vs. upregulation. more biologically relevant modules. 
```


```{r, generation of topological overlap}
#Topological overlap matrix - second step that cleans up the adjacency network. Inferences become more interpretable.
TOM <- TOMsimilarity(adjacency,
                     TOMType = "signed",
                     verbose = 5)
dissTOM <- 1-TOM
# TOM is a measure of similarity. But for clustering you want to see dissimilarity of clusters. So that's why you run dissTOM
rm(adjacency) # may need to delete adjacency, TOM to clear up vector memory
```


```{r, Gene clustering with TOM-based dissimilarity plot}
geneTree_ofav <- flashClust(as.dist(dissTOM), method = "average") 
plot(geneTree_ofav, labels = FALSE, xlab="", sub="", main = "Gene clustering on TOM-based dissimilarity")
```


```{r, Module Identification}
# minModuleSize = 30 because Mike prefers large modules that lend themselves to enrichment tests with GO_MWU for easier interpretation.
minModuleSize <- 30 #number of genes. this is arbitrary. want to comprise a network. 30, 50, 100. up to you

# Module identification using dynamic tree cut, with cutHeight = 0.99 and deepSplit = 2 based on the recommendations of the WGCNA developers
#looking at tree of genes similarity. Cutting it at a certain height to choose which genes to map together into modules. 
dynamicMods <- cutreeDynamic(dendro = geneTree_ofav,
                             distM = dissTOM,
                             cutHeight = 0.99, #pretty standard setting
                             deepSplit = 2,
                             pamRespectsDendro = FALSE,
                             minClusterSize = minModuleSize)
table(dynamicMods) #label zero is unassigned genes. labels 1-20 are the modules
```

```{r}
# Convert numeric lables into colors
dynamicColors = labels2colors(dynamicMods)
table(dynamicColors)
# Plot the dendrogram and colors underneath
sizeGrWindow(8,6)
plotDendroAndColors(geneTree_ofav, dynamicColors, "Dynamic Tree Cut",
dendroLabels = FALSE, hang = 0.03,
addGuide = TRUE, guideHang = 0.05,
main = "Gene dendrogram and module colors")
```

```{r}
# Calculate eigengenes
MEList = moduleEigengenes(datExpr_ofav, colors = dynamicColors)
MEs = MEList$eigengenes
# Calculate dissimilarity of module eigengenes
MEDiss = 1-cor(MEs);
# Cluster module eigengenes
METree = hclust(as.dist(MEDiss), method = "average");
# Plot the result
sizeGrWindow(7, 6)
plot(METree, main = "Clustering of module eigengenes",
xlab = "", sub = "")

#this DissThreshold corresponds to an 85% correlation (so it's pretty stringent? in the WGCNA tutorial online they use a 0.25 threshold)
MEDissThres = 0.15

# Plot the cut line into the dendrogram
abline(h=MEDissThres, col = "red")

# Call an automatic merging function
# merge cutHeight = 0.3 because I want to group together modules with >85% similar module eigengene expression
mergedMods <- mergeCloseModules(datExpr_ofav, dynamicColors, cutHeight = 0.15, verbose = 5)

# The merged module colors
mergedColors <- mergedMods$colors
table(mergedColors) %>% as.data.frame() %>% write_csv("merged_colors_num_of_genes_ineachmodule.csv")

MEList <- moduleEigengenes(datExpr_ofav, colors = mergedColors)
mergedMEs <- MEList$eigengenes
#this is recalculating eigengenes for merged modules
```

```{r}
sizeGrWindow(12, 9)
#pdf(file = "Plots/geneDendro-3.pdf", wi = 9, he = 6)
plotDendroAndColors(geneTree_ofav, cbind(dynamicColors, mergedColors),
c("Dynamic Tree Cut", "Merged dynamic"),
dendroLabels = FALSE, hang = 0.03,
addGuide = TRUE, guideHang = 0.05)
#dev.off()
```

```{r, module eigengenes and colors}
# Rename to moduleColors
moduleColors = mergedColors
# Construct numerical labels corresponding to the colors
colorOrder = c("grey", standardColors(50));
moduleLabels = match(moduleColors, colorOrder)-1;
MEs = mergedMEs;
# Save module colors and labels for use in subsequent parts
#save(MEs, moduleLabels, moduleColors, geneTree_ofav, file = "modules_ofavgenetree.RData")
```

```{r ME_clustering based on module eigengenes}
datME <- MEs
dissimME <- (1-t(cor(datME, method="p")))/2
hclustME <- flashClust(as.dist(dissimME), method="average" )
# Plot the eigengene dendrogram
par(mfrow=c(1,1))
plot(hclustME, main="Clustering tree based on the module eigengenes")
```

```{r uniq_modules}
# Extract all unique modules
uniqModules <- unique(colnames(MEs))
# set in order according to ME clustering dendogram
hclustME$order
uniqModules <- uniqModules[hclustME$order]
#
uniqModules
# create ordered factor for downstream analysis
modules_factor <- factor(levels = uniqModules[hclustME$order], ordered = TRUE)

dt <- data.frame(uniqModules[hclustME$order]) #ordered from 
```

### Module eigengene correlations
```{r}
# Tidy sample metadata into a format WGCNA can use for correlations
ofav_mote_traits <- ofav_metadata %>% 
  select(Working.Label, Correct_treatment, Experiment)
#
for (i in 1:nrow(ofav_mote_traits)){
if (ofav_mote_traits$Correct_treatment[i] == "control") {
  ofav_mote_traits$control[i] <- "1"
} else {
  ofav_mote_traits$control[i] <- "0"
}
}
# 
for (i in 1:nrow(ofav_mote_traits)){
if (ofav_mote_traits$Correct_treatment[i] == "diseased") {
  ofav_mote_traits$disease[i] <- "1"
} else {
  ofav_mote_traits$disease[i] <- "0"
}
}


ofav_mote_traits %>% select(!Working.Label:Experiment)-> ofav_mote_traits


#convert genotype to numeric for correlation
ofav_genotype <- ofav_metadata %>% mutate(Exp_frag_genet = as.numeric(as.factor(Exp_frag_genet))) %>% 
  select(Exp_frag_genet)
```

###Quantifying moduleâ€“trait associations
```{r}
# Define numbers of genes and samples
nGenes = ncol(datExpr_ofav);
nSamples = nrow(datExpr_ofav);
# Recalculate MEs with color labels
MEs0 = moduleEigengenes(datExpr_ofav, moduleColors)$eigengenes
MEs = orderMEs(MEs0)
moduleTraitCor = cor(MEs, ofav_mote_traits$disease, use = "p");
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples);

disease = as.data.frame(ofav_mote_traits$disease);
names(disease) = "disease"

sizeGrWindow(14,6)
# Will display correlations and their p-values
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",
signif(moduleTraitPvalue, 1), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor)
par(mar = c(3, 8.5, 3, 3));
# Display the correlation values within a heatmap plot
labeledHeatmap(Matrix = moduleTraitCor,
xLabels = names(disease),
xLabelsAngle = 0,
xLabelsAdj = 0.5,
yLabels = names(MEs),
ySymbols = names(MEs),
colorLabels = FALSE,
colors = blueWhiteRed(50),
textMatrix = textMatrix,
setStdMargins = FALSE,
cex.text = 0.3,
zlim = c(-1,1),
main = paste("Module-trait Relationships"))


moduleTraitCor_genotype = cor(MEs, ofav_genotype$Exp_frag_genet, use = "p")
moduleTraitPvalue_genotype = corPvalueStudent(moduleTraitCor_genotype, nSamples);

sizeGrWindow(14,6)
# Will display correlations and their p-values
textMatrix = paste(signif(moduleTraitCor_genotype, 2), "\n(",
signif(moduleTraitPvalue_genotype, 1), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor_genotype)
par(mar = c(3, 8.5, 3, 3));
# Display the correlation values within a heatmap plot
labeledHeatmap(Matrix = moduleTraitCor_genotype,
xLabels = names(ofav_genotype),
xLabelsAngle = 0,
xLabelsAdj = 0.5,
yLabels = names(MEs),
ySymbols = names(MEs),
colorLabels = FALSE,
colors = blueWhiteRed(50),
textMatrix = textMatrix,
setStdMargins = FALSE,
cex.text = 0.3,
zlim = c(-1,1),
main = paste("Module-trait Relationships"))
```
###Gene Significance and Module Membership - quantifying associations between gene and trait (disease exposure)
```{r}
# Define variable weight containing the weight column of datTrait
disease = as.data.frame(ofav_mote_traits$disease);
names(disease) = "disease"
# names (colors) of the modules
modNames = substring(names(MEs), 3)
geneModuleMembership = as.data.frame(cor(datExpr_ofav, MEs, use = "p"));
MMPvalue = as.data.frame(corPvalueStudent(as.matrix(geneModuleMembership), nSamples));

names(geneModuleMembership) = paste("MM", modNames, sep="");
names(MMPvalue) = paste("p.MM", modNames, sep="");
geneTraitSignificance = as.data.frame(cor(datExpr_ofav, disease, use = "p"));
GSPvalue = as.data.frame(corPvalueStudent(as.matrix(geneTraitSignificance), nSamples));
names(geneTraitSignificance) = paste("GS.", names(disease), sep="");
names(GSPvalue) = paste("p.GS.", names(disease), sep="");

rownames_to_column(geneTraitSignificance, var="Locus") -> geneTraitSignificance
rownames_to_column(geneModuleMembership, var="Locus") -> geneModuleMembership

merged_GS_MM <- inner_join(geneTraitSignificance, geneModuleMembership, by = "Locus")

merged_GS_MM %>% 
tidyr::gather(module, module_membership, MMpurple:MMgrey) %>%
  filter(module== "MMtan" | module == "MMturquoise") %>% 
ggplot(aes(module_membership, GS.disease, color = module)) +
  geom_point(size = 2, alpha = 1/100) +
  geom_smooth(color="black", method = lm, se = T, fill = "grey") +
  theme_classic() +
  ylim(c(-1,1)) +
  xlim(c(-1,1)) +
  facet_wrap(~module, scales = "free") +
  scale_color_manual(values = c("tan", "turquoise")) +
  theme(legend.position = "none") +
  labs(x="Module Membership",
       y="Gene Significance for Disease")
# ggsave("MM_GS_tan_turquoise.png")

```


### Measure Module Significance as Average Gene Significance
```{r}
GS1=as.numeric(cor(disease,datExpr_ofav, use="p"))
GeneSignificance=abs(GS1)
# Next module significance is defined as average gene significance.
ModuleSignificance=tapply(GeneSignificance, moduleColors, mean, na.rm=T)

sizeGrWindow(8,7)
par(mfrow = c(1,1))
plotModuleSignificance(GeneSignificance,moduleColors)
```

### hub genes
```{r modules_genes}
# Generate GeneIDs
Genes <- colnames(datExpr_ofav)

#hub genes for each module
tophub <- chooseTopHubInEachModule(datExpr_ofav, moduleColors, power = 10)

tophub <- as.data.frame(tophub)
tophub$Gene.ID <- tophub$tophub
tophub %>% 
  rename(Gene_ID = Gene.ID) -> tophub
tophub %>% rownames_to_column(var="module") ->tophub

ofav_annot_DEG <- read.csv(here("ofav_annotated_DEGresults_allgenes.csv")) #this table is linking gene IDs to protein IDs and protein names. the ID matches from the NCBI refseq version of the Ofav genome, created through their automatic annotation pipeline

ofav_annot_DEG %>% 
  dplyr::select(Gene_ID, Gene.Annotation) %>% 
  right_join(tophub) %>% 
  as.data.frame() %>% 
 select(!Gene_ID) -> tophub_annotated
             
View(tophub_annotated)

write_csv(tophub_annotated, "tophub_annotated.csv")
```


```{r, genes for each module annotated}
# Output genes and annotations for each module
for (module in unique(moduleColors))
{
# Select module genes
inModule <- (moduleColors == module)
# Get gene IDs
modGeneIDs <- Genes[inModule] #this is the correct set of gene IDs!
# Write gene annotations into a file
fileName = paste(module, ".csv", sep="");
module_annotation <- ofav_annot_DEG[ofav_annot_DEG$Gene_ID %in% modGeneIDs, ]
write.csv(module_annotation, file = fileName)
}
```

Identifying genes with highest module membership in tan and turquoise modules 
```{r}

tanmodule <- read_csv("../data and tables for revisions/modules/tan.csv")
geneModuleMembership %>% 
  select(Locus, MMtan) %>% 
  arrange(desc(MMtan)) %>% 
  left_join(ofav_annot_DEG %>% rename(Locus = Gene_ID) %>% select(Locus, Gene.Annotation)) %>% 
  arrange(desc(MMtan)) %>% 
  right_join(tanmodule %>% rename(Locus = Gene_ID)) 
  #write_csv("tan_modulemembership_annotated.csv")
  
#top gene = LOC110069306 with module membership value of 0.9608815, annotation is uncharacterized protein LOC110069306 isoform X2.
#second highest gene (which is annotated), LOC110062260 (0.9550084)	 -- DDB1- and CUL4-associated factor 11-like
  
turquoisemodule <- read_csv("../data and tables for revisions/modules/turquoise.csv")
geneModuleMembership %>% 
  select(Locus, MMturquoise) %>% 
  arrange(desc(MMturquoise)) %>% 
  left_join(ofav_annot_DEG %>% rename(Locus = Gene_ID) %>% select(Locus, Gene.Annotation)) %>% 
  arrange(desc(MMturquoise))%>% 
  right_join(turquoisemodule %>% rename(Locus=Gene_ID))
  #write_csv("turquoise_modulemembership_annotated.csv")
#top gene: LOC110065529	0.9734237	golgin subfamily B member 1-like isoform X2
```


###Plot the Eigengene adjacency heatmap matrix 
```{r} 
MET = orderMEs(cbind(MEs, disease))
par(cex = 1.0)
sizeGrWindow(6,6);
plotEigengeneNetworks(MET, "Eigengene adjacency heatmap", marHeatmap = c(3,4,2,2),
plotDendrograms = FALSE, xLabelsAngle = 90)
```


